### PR TITLE
Add eval-script accessible replay JS API.

### PR DESCRIPTION
**Pull request by kannanvijayan**
Adds the global `__RECORD_REPLAY__`, accessible to scripts sent by `eval*` commands, and provides two methods:

`getOrCreateProtocolId(obj)` - get the protocol id for an object. `getObjectFromProtocolId(rrpId)` - get an object corresponding to a protocol id.

For #RUN-1177 (https://linear.app/replay/issue/RUN-1177).

### CODE REVIEW

**Comment by kannanvijayan** [comment]
Build for this is still running, and there a bunch of passing and failing tests, but that's immaterial: if this patch is bad, it'll cause every single test to fail.. since changes are at initialization only and the API functions added are otherwise unexercised.

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 1]
```diff
@@ -2892,6 +2892,19 @@ __RECORD_REPLAY_ARGUMENTS__.internal = {
   updateNextStackingContextId: (f) => { gNextStackingContextId = f(gNextStackingContextId); },
 };
 
+/** ###########################################################################
+ * Export JS API methods via `__RECORD_REPLAY__`.
+ * This is officially available for scripts in `eval*` commands to use.
+ * ##########################################################################*/
+Object.assign(__RECORD_REPLAY__, {
+  getOrCreateProtocolId(obj) {
```
Comment (loganfsmyth) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:2900]: The fact that the ID is created on the fly is probably an implementation detail, so if we're doing this I'd be more for `getProtocolObjectId` or something. Maybe
* `getProtocolIdForObject`
* `getObjectFromProtocolId`
?

Comment (loganfsmyth) [reply third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:2900]: I still have mixed feelings about taking this approach as an official API, but I recognize that it's probably fine as a first pass.

Comment (kannanvijayan) [reply third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:2900]: Yeah, I had the same thought about the naming.  I'll do that.

**Review by klochek — APPROVED** [review]

**Pull request merged**